mongodbatlas/fw_resource_mongodbatlas_search_deployment.go

```diff
@@ -304,7 +322,7 @@ func searchDeploymentRefreshFunc(ctx context.Context, projectID, clusterName str
 				return "", retrystrategy.RetryStrategyDeletedState, nil
 			}
 			if resp.StatusCode == 503 {
-				return "", retrystrategy.RetryStrategyPendingState, nil
+				return "", retrystrategy.RetryStrategyUpdatingState, nil
```

Comment:
Detected a bug here thanks to test case. Not very relevant, simply a logic that continues the retry logic if a "Service Unavailable" is returned from API.

Reply:
ah, so nice to read this - music to our ears :)

Reply:
What was the strategy before? the name `RetryStrategyPendingState` is not super clear 🤔

Reply:
`RetryStrategyPendingState` is used for encryption at rest resource, but is not a valid state for the search deployment. The correct state for representing that a change is in progress is with `RetryStrategyUpdatingState`.